### ISSUE TITLE
Firmware update fails on LiteVNA64 flashed with LiteVNA62 FW

### ISSUE THREAD

**Issue by fiveangle** [joeqsmith/NanoVNA-Solver64-#1]
Not that big of a deal because people shouldn't be flashing the incorrect FW to a device, but but in case this bug might be sign of a more sinister bug that could affect a normal FW flash at some point, below is the repro steps:

1. Flash incorrect FW LiteVNA62-220228.bin to LiteVNA64 device using Solver64 (https://github.com/joeqsmith/NanoVNA-Solver64-/commit/3a84660c9c1808fd37d57194ab844d1a36fcb692); observe successful flash and LiteVNA64 screen displays smaller/mirrored screen output due to incorrect FW
2. Attempt to flash correct FW LiteVNA64-220228.bin to the device using the same Solver64; observe progress bar immediately completes but FW is not updated; there is no way to flash the correct FW to the device using Solver64
3. Attempt to flash the device using NanoVNA-App; observe correctly detects device in DFU mode and correctly flashes to the requested appropriate FW 

Thanks Joe !

**Comment by joeqsmith** [comment]
Once my software has programmed the VNA it will reset it and then disconnect.  After step 1, you are running the application.  You never power cycled the VNA with the selector pressed to place it back into DFU.  You then never reconnect to the device with my software.  Instead you proceed to step 2 and tell the software to program a VNA that is not in DFU mode or connected.  It is happy to do this for you and as you notice, programming a non existing VNA happens fast.   

Of course there is a way to flash the correct firmware.  You just need to follow the same steps.  Place the device into DFU mode, connect to it and download the new firmware.   

https://www.eevblog.com/forum/rf-microwave/nanovna-custom-software/msg4106665/#msg4106665

**Comment by fiveangle** [comment]
Joe, you didn't follow the repro steps.  If you did, you would have seen that if your assumption were correct, then there is no way I would have completed step 1:

> 1. Flash incorrect FW LiteVNA62-220228.bin to LiteVNA64 device using Solver64 (https://github.com/joeqsmith/NanoVNA-Solver64-/commit/3a84660c9c1808fd37d57194ab844d1a36fcb692); observe successful flash and LiteVNA64 screen displays smaller/mirrored screen output due to incorrect FW

As you presume the unit was correctly prepped to "flash" in step 1 , you should also presume that the necessary steps were completed in step 2 which I stated the same process:
"2. Attempt to flash correct FW LiteVNA64-220228.bin to the device using the same Solver64; "

I can assure you Step 1 completed as I stated and Step 2 did so as well using reasonable expectations (cycling into DFU mode, etc). 

If however, you are telling me you attempted to perform the steps as written with reasonable assuptions and were successful,  I will retry (for the 6th time - I attempted 5 complete separate times without success even after returning to correct FW using Saver each time, before giving up and resolving that there is a bug.

**Comment by fiveangle** [comment]
Since Joe wasn't convinced, I thought I should rule out my system so I copied the (HUGE) NI runtime binaries over to my old Win10 laptop and unfortunately (or fortunately, for Solver) could not repro the issue.  This would point to a system issue but it didn't make sense why Solver could flash the FW at all on the first system, so I moved from the front USB3 port I had been using (connected to a bridge chip on the MSI mobo I have) and to one of the native USB2 ports direct on the rear of the mobo, and flashing back and forth was successful.  I then moved to the rear mobo native USB3.0 ports and flashing back and forth was successful.  So I then went back to the bridge-connected front USB3 port and it was successful 3 times, then failed on the 4th.  I then moved back to Saver and flashed successfully back and forth 5 times each without failure.  So it seems there may be some intermittent/rarish timing issue with the MSI USB3 bridge chip in the B450 based mobo I have.  What I can't explain is why it failed 5 times in succession last night but worked 3 times in a row tonight.  Perhaps sunspots (I was also very successful with CB skip last night around the same time ;)

Regardless, good to find out this wasn't a bug in Solver. If you encounter any flash problems, obviously ensure you are following the correct steps as Joe states, but try a mobo-direct USB port just in case.
